src/Servers/Kestrel/Core/src/Internal/Infrastructure/TransportManager.cs

```diff
@@ -47,15 +49,28 @@ public async Task<EndPoint> BindAsync(EndPoint endPoint, ConnectionDelegate conn
             return transport.EndPoint;
         }
 
-        public async Task<EndPoint> BindAsync(EndPoint endPoint, MultiplexedConnectionDelegate multiplexedConnectionDelegate, EndpointConfig? endpointConfig)
+        public async Task<EndPoint> BindAsync(EndPoint endPoint, MultiplexedConnectionDelegate multiplexedConnectionDelegate, ListenOptions listenOptions)
         {
             if (_multiplexedTransportFactory is null)
             {
                 throw new InvalidOperationException($"Cannot bind with {nameof(MultiplexedConnectionDelegate)} no {nameof(IMultiplexedConnectionListenerFactory)} is registered.");
             }
 
-            var transport = await _multiplexedTransportFactory.BindAsync(endPoint).ConfigureAwait(false);
-            StartAcceptLoop(new GenericMultiplexedConnectionListener(transport), c => multiplexedConnectionDelegate(c), endpointConfig);
+            var features = new FeatureCollection();
+
+            if (listenOptions.HttpsOptions != null)
+            {
+                // TODO Set other relevant values on options
+                var sslServerAuthenticationOptions = new SslServerAuthenticationOptions
+                {
+                    ServerCertificate = listenOptions.HttpsOptions.ServerCertificate
```

Comment:
There are a lot of HTTPS options available on HttpsConnectionAdapterOptions.

Which are valid for HTTP/3 and msquic?

Reply:
Assume all of them are relevant until proven otherwise. If something isn't supported by MSQuic then we need a good explanation of why and it should be documented on those options.

Reply:
Issue to support more: https://github.com/dotnet/aspnetcore/issues/30857